content/assignments/minishell/checker/_test/run_test.sh

```diff
@@ -0,0 +1,244 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+
+# ----------------- General declarations and util functions ------------------ #
+
+exec_name="mini-shell"
+ref_name="bash"
+ref_name_alt="dash"
+
+MAIN_TEST_DIR="_test/outputs"
+INPUT_DIR="_test/inputs"
+REFS_DIR="_test/refs"
+LOG_FILE="/dev/null"
+export max_points=100
+TEST_TIMEOUT=30
+
+TEST_LIB=_test/test_lib.sh
+export CHECK_SKIP='parser.\(\(tab\|yy\).\(c\|h\)\|h\)$'
+
+# Load the lib functions.
+if ! [ -e "$TEST_LIB" ]; then
+	echo "Test library not found. Check \$TEST_LIB ($TEST_LIB)"
+	exit 1
+fi
+source "$TEST_LIB"
+
+export SHELL_PROMPT="> "
+
+# Change this of you want to keep the logs after execution.
+DO_CLEANUP=no
+
+BUFFERING_WRAPPER=""
+if command -v stdbuf &> /dev/null; then
+	# stdbuf may not be installed on cygwin.
+	BUFFERING_WRAPPER="stdbuf -i 0"
+fi
+
+USE_VALGRIND=yes
+VALGRIND_LOG="valgrind.log"
+
+# ----------------- Init and cleanup tests ----------------------------------- #
+
+# Initializes a test.
+init_test()
+{
+
+	if ! [ -e "$exec_name" ]; then
+		echo "$exec_name not found! Cannot run the test"
+		exit 1
+	fi
+
+	if [ "$USE_VALGRIND" = "yes" ]; then
+		which valgrind &> /dev/null
+		if [ $? -eq 0 ]; then
+			exec_name="valgrind --leak-check=full \
+			                    --show-leak-kinds=all \
+					    --child-silent-after-fork=yes \
+					    --track-fds=yes \
+					    --log-file=$VALGRIND_LOG \
+					    $exec_name"
+		else
+			echo "valgrind is not installed."
+		fi
+	fi
+
+	# Generates random file as test file.
+	TEST_NAME=$(printf "test_%02d" "$test_index")
+	OUT_DIR="${TEST_NAME}_out"
+	REF_DIR="${TEST_NAME}_ref"
+	IN_FILE="${TEST_NAME}.in"
+	REF_FILE="${TEST_NAME}.ref"
+	OUT_FILE="${TEST_NAME}.out"
+
+	local input_basename="${INPUT_DIR}/${TEST_NAME}"
+	[ -f "${input_basename}.${OS_PLATFORM}.txt" ] && \
+		input_basename="${input_basename}.${OS_PLATFORM}" || \
+		input_basename="${input_basename}"
+
+	# Copy input files in the testing directory.
+	cp "${input_basename}.txt" "${MAIN_TEST_DIR}/${IN_FILE}" &> "$LOG_FILE"
```

Comment:
`cp` will print nothing by default (i.e. `-v` is not used), so the redirect does nothing. Also, if the `LOG_FILE` variable were a proper file, it would be emptied by the non-appending `>` redirect.